src/source/mod.rs

```diff
@@ -306,20 +306,68 @@ pub trait NpmResolver: fmt::Debug {
 pub fn load_data_url(
   specifier: &ModuleSpecifier,
 ) -> Result<Option<LoadResponse>, anyhow::Error> {
-  let url = DataUrl::process(specifier.as_str())
-    .map_err(|_| anyhow!("Unable to decode data url."))?;
-  let (bytes, _) = url
-    .decode_to_vec()
-    .map_err(|_| anyhow!("Unable to decode data url."))?;
-  let mut headers: HashMap<String, String> = HashMap::with_capacity(1);
-  headers.insert("content-type".to_string(), url.mime_type().to_string());
+  let data_url = RawDataUrl::parse(specifier)?;
+  let (bytes, headers) = data_url.into_bytes_and_headers();
   Ok(Some(LoadResponse::Module {
     specifier: specifier.clone(),
     maybe_headers: Some(headers),
     content: Arc::from(bytes),
   }))
 }
 
+#[derive(Debug, Clone)]
+pub struct RawDataUrl {
+  pub mime_type: String,
+  pub bytes: Vec<u8>,
+}
+
+impl RawDataUrl {
+  pub fn parse(specifier: &ModuleSpecifier) -> Result<Self, Error> {
+    let url = DataUrl::process(specifier.as_str())
+      .map_err(|_| anyhow!("Unable to decode data url."))?;
+    let (bytes, _) = url
+      .decode_to_vec()
+      .map_err(|_| anyhow!("Unable to decode data url."))?;
+    Ok(RawDataUrl {
+      mime_type: url.mime_type().to_string(),
+      bytes,
+    })
+  }
+
+  pub fn charset(&self) -> Option<&str> {
+    get_mime_type_charset(&self.mime_type)
+  }
+
+  pub fn media_type(&self) -> MediaType {
+    let mut content_types = self.mime_type.split(';');
+    let Some(content_type) = content_types.next() else {
+      return MediaType::Unknown;
+    };
+    MediaType::from_content_type(
+      &ModuleSpecifier::parse("data:image/png;base64,").unwrap(),
```

Comment:
Is this just to satisfy the type system? Can there be a situation where this will return wrong value? (I guess it will return `Unknown` if `content_type` is not recognized)

Reply:
Ah, yeah I'll add a comment, todo, and open an issue in media_type to improve this. Yeah, it just gets ignored.

Reply:
Actually, I'll just add a comment here. This is extremely edge casey and in 99% of cases someone should be providing a specifier to this.